projects/igniteui-angular/src/lib/grids/group-area.component.ts

```diff
@@ -0,0 +1,297 @@
+import { Component, ElementRef, EventEmitter, Input, Output, TemplateRef, ViewChild, AfterContentInit } from '@angular/core';
+import { IBaseChipEventArgs, IChipClickEventArgs, IChipKeyDownEventArgs } from '../chips/chip.component';
+import { IChipsAreaReorderEventArgs } from '../chips/chips-area.component';
+import { IGroupingExpression } from '../data-operations/grouping-expression.interface';
+import { SortingDirection } from '../data-operations/sorting-expression.interface';
+import { IgxColumnComponent } from './columns/column.component';
+import { IgxGridComponent } from './grid/grid.component';
+import { IgxColumnMovingDragDirective } from './moving/moving.drag.directive';
+import { IgxTreeGridComponent } from './tree-grid/tree-grid.component';
+
+/**
+ * Provides a pre-configured column grouping component for the tree grid.
+ *
+ * @example
+ * ```html
+ *  <igx-group-area></igx-group-area>
+ * ```
+ */
+@Component({
+    selector: 'igx-group-area',
+    templateUrl: './group-area.component.html'
+})
+export class IgxGroupAreaComponent implements AfterContentInit {
+    @Input()
+    public grid: IgxGridComponent | IgxTreeGridComponent;
+
+    @Input()
+    public get groupingExpressions(): IGroupingExpression[] {
+        return this._groupingExpressions;
+    }
+    public set groupingExpressions(value: IGroupingExpression[]) {
+        if (this.grid instanceof IgxTreeGridComponent) {
+            const groupingExpressionsCount = this._groupingExpressions.length;
+            this.grid.sortingExpressions.splice(0, groupingExpressionsCount);
```

Comment:
I believe we should try to decouple the group area implementation from the grid's concrete implementation. More specifically, I'd rather abstract away this.grid.sortingExpressions, this.grid.columnList, etc.